worker/server.go

```diff
@@ -29,22 +29,21 @@ func (s Server) StartExtractingLinks(ctx context.Context, request *proto.StartEx
 	return &empty.Empty{}, nil
 }
 
-// Start grpc server for worker
+// Start will start a grpc server for worker
 func (s Server) Start(port int) error {
-	address := fmt.Sprintf(":%d", port)
-	lis, err := net.Listen("tcp", address)
+	gRPCService, err := service.
+		NewGRPCBuilder("Worker").
+		RegisterHandler(func(server *grpc.Server) {
+			proto.RegisterWorkerServer(server, &s)
+		}).
+		Build()
 	if err != nil {
-		return err
+		panic(err)
 	}
-	sv := grpc.NewServer()
-	proto.RegisterWorkerServer(sv, &s)
-	fmt.Printf("Worker started at %d\n", port)
 
-	err = s.worker.Init("localhost", port)
-	if err != nil {
-		return err
```

Comment:
should this be changed to panic to be consistent with line 41

Reply:
Returning error is what we wanted.